chugbot/attendance/index.php

```diff
@@ -0,0 +1,238 @@
+fillId2Name(null, $blockId2Name, $dbErr, "block_id", "blocks");
+fillId2Name(null, $groupId2Name, $dbErr, "group_id", "chug_groups");
+fillId2Name(null, $edahId2Name, $dbErr, "edah_id", "edot");
+fillId2Name(null, $bunkId2Name, $dbErr, "bunk_id", "bunks");
+
+
+setup_camp_specific_terminology_constants();
+
+echo headerText("Attendance Admin");
+
+// update saved attendance
+if ($_SERVER["REQUEST_METHOD"] == "POST") {
+    // determine what is being updated/set
+    if(test_post_input("form") == "active-blocks") {
+        $localErr = "";
+        $dbc = new DbConn();
+        // list of applicable chug group ids
+        $groupIds = array_keys($groupId2Name);
+
+        // build SQL
+        $anySet = FALSE;
+        $sql = "UPDATE chug_groups SET active_block_id = ( CASE ";
+        foreach($groupIds as $id) {
+            $block = test_post_input($id);
+            if (!empty($block)) {
+                $sql .= "WHEN (group_id = $id) THEN $block ";
+                $anySet = TRUE;
+            }
+        }
+        $sql .= "ELSE (NULL) END )";
+        // if all are set to null, override built SQL and just clear column
+        if(!$anySet) {
+            $sql = "UPDATE chug_groups SET active_block_id = NULL";
+
+        }
+        
+        // run query
+        $result = $dbc->doQuery($sql, $localErr);
+        if ($result == false) {
+            echo dbErrorString($sql, $localErr);
+            exit();
+        }
+        else {
+            $successMessage = "<div class=\"col-md-6 offset-md-3\"><div class=\"alert alert-success alert-dismissible fade show m-2\" role=\"alert\">" . 
+            "<button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"alert\" aria-label=\"Close\"></button><h5>Active " . ucfirst(block_term_plural) .
+            " Successfully Updated</h5>People taking and viewing attendance will now see the " . chug_term_singular . " for the newly set " . block_term_plural . "</div></div>";
+            echo $successMessage;
+    
+        }
+        
+    }
+    else if (test_post_input("form") == "purge") {
+        // delete old records
+
+        // 1: verify date is valid form
+        $dateInput = test_post_input("date");
+        $fullDate = strtotime($dateInput);
+        $date = date('Y-m-d', $fullDate);
+        if($fullDate == "") {
+            echo "<div class=\"col-md-6 offset-md-3\"><div class=\"alert alert-danger alert-dismissible fade show m-2\" role=\"alert\">" . 
+            "<button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"alert\" aria-label=\"Close\"></button><h5><strong>Error:</strong> " . 
+            "Invalid Date</h5>An invalid date was provided. Please try again to purge old attendance records</div></div>";
+            exit();
+        }
+        else {
+            // 2: do deletions
+            $localErr = "";
+            $dbc = new DbConn();
+            // 2a: delete attendance records
+            $sql = "DELETE FROM attendance WHERE date < '$date'";
```

Comment:
Yes - I was thinking it was something which could be done at the end of the summer when old attendance records don't really matter. I put a decent number of safeguards into place to avoid accidentally submitting it - the "purge" button is disabled until a valid date is entered, and that button leads to a big alert popup (which has the button to actually submit it also disabled until there is a valid date and the popup appears). Again, I wouldn't be opposed to making an option more specific, just not sure what the main use case would be from the admin side.

Reply:
Sounds good! I agree- if you're deleting attendance records, it's unlikely you'd want to keep a subset. In general, I favor keeping the user interface simple for admins, so this is fine.